### PR TITLE
chore(n8n): update image to 2.22.3

### LINKED ISSUE #319: chore(n8n): upstream image update available (2.21.3 → 2.22.3)
## Upstream Image Update Available

| Field | Value |
|---|---|
| **Chart** | `n8n` |
| **Current image** | `docker.io/n8nio/n8n:2.21.3` |
| **New version** | `2.22.3` |
| **Registry** | Docker Hub |

### Update Checklist

- [ ] Verify new tag exists and is stable: `docker pull docker.io/n8nio/n8n:2.22.3`
- [ ] Review upstream changelog/release notes for breaking changes
- [ ] Update `values.yaml` image tag
- [ ] Update `appVersion` in `Chart.yaml`
- [ ] Run `helm lint charts/n8n --strict`
- [ ] Run `helm unittest charts/n8n`
- [ ] Test on local k3d cluster
- [ ] Update site documentation if needed (GR-007)

---
_This issue was automatically created by the [Upstream Watch](https://github.com/helmforgedev/charts/actions/workflows/upstream-watch.yml) workflow._

### PR DESCRIPTION
**Pull request by mberlofa**
Closes #319.

## Summary
- Update the n8n chart to image/appVersion 2.22.3.
- Standardize Gateway API values on `gatewayAPI` while keeping the legacy `gateway` block compatible.
- Remove the tracked n8n `Chart.lock` and ignore chart lock files per HelmForge GR-062.

## Validation
- `helm dependency build charts/n8n`
- `helm dependency list charts/n8n`
- `helm lint charts/n8n`
- `helm lint --strict charts/n8n`
- `helm template n8n-test charts/n8n` and all n8n CI values files
- `helm unittest charts/n8n`: 9 suites, 53 tests passed
- `kubeconform` strict on all n8n CI values: 0 invalid, 0 errors
- `ah lint charts/n8n`: 65 packages, 0 errors
- Kubescape MITRE, NSA, SOC2: overall 84, Resource Summary 83.50%
- k3d runtime on `k3d-helmforge-tests-wsl`: pod Ready 1/1, PVC Bound, `/healthz` 200, UI 200, no non-Normal events; release and namespace cleaned

## Upstream
- `docker.io/n8nio/n8n:2.22.3` exists for linux/amd64 and linux/arm64.
- GitHub release `n8n@2.22.3` was published on 2026-05-25 and includes the SSO claims/role mapping diagnostics bug fix.
- GitHub marks this upstream release as prerelease; this PR uses it because issue #319 explicitly requested 2.22.3 and the official Docker tag is available.

## Site sync
- `/docs/charts/n8n#values` is queued for the site repository update after the chart issue batch, per the requested cross-repo workflow.

### CODE REVIEW

File: charts/n8n/templates/httproute.yaml [hunk 1]
```diff
@@ -1,33 +1,41 @@
 {{/* SPDX-License-Identifier: Apache-2.0 */}}
-{{- if .Values.gateway.enabled }}
-{{- if not .Values.gateway.gatewayName }}
-  {{- fail "gateway.gatewayName is required when gateway.enabled=true" }}
+{{- $gatewayAPI := default dict .Values.gatewayAPI }}
+{{- $legacyGateway := default dict .Values.gateway }}
+{{- if or .Values.gatewayAPI.enabled .Values.gateway.enabled }}
+{{- $gatewayName := default (default "" $legacyGateway.gatewayName) $gatewayAPI.gatewayName }}
```
[the remark this answers is not in the document]
Comment (mberlofa) [reply charts/n8n/templates/httproute.yaml]: Addressed in 364efde. The HTTPRoute now selects either `gatewayAPI.*` or legacy `gateway.*` values based on the enabled block instead of falling back across both. When `gatewayAPI.enabled=true`, a missing `gatewayAPI.gatewayName` now fails fast even if legacy `gateway.gatewayName` is still present. Added unit tests for the migration edge case and for Gateway API precedence. Validation: `helm lint charts/n8n`, `helm template`, `helm install --dry-run=client --debug`, and `helm unittest charts/n8n` all pass.
